android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java

```diff
@@ -161,9 +163,28 @@ private void updateState() {
     }
 
     private void updateStatus() {
+        GURL address = getCurrentHostHttpAddress();
+        org.chromium.url.internal.mojom.Origin hostOrigin =
+                new org.chromium.url.internal.mojom.Origin();
+        hostOrigin.scheme = address.getScheme();
+        hostOrigin.host = address.getHost();
+        hostOrigin.port = 0;
+        if (address.getPort().isEmpty()) {
+            if (hostOrigin.scheme.equals("http")) {
```

Comment:
I don't really get why we cannot pass `.spec()` from GURL to a mojom call and after that on c++ side convert it to GURL?

Reply:
The idea is to switch from GURL and std::string to url::Origin to increase type-safety in origin contexts. For wallet bubble there is actually no such manual origin construction needed as it comes from browser via BraveWalletServiceObserver::OnActiveOriginChanged. Not sure if that is somehow easily applicable for android

Reply:
ok, let it be for now!